File: src/UserService.cs
using Raymaker.RulesEngine.Application.Model;

namespace Raymaker.RulesEngine.Application
{
    public class UserService : IUserService
    {
        private readonly IUserRepository userRepository;
        private readonly UserValidator userValidator;

        public UserService() : this(new UserRepository()) { }

        public UserService(IUserRepository userRepository)
        {
            this.userRepository = userRepository;
            this.userValidator = new UserValidator();
        }

        public User GetUser(string userName)
        {
            return this.userRepository.GetUser(userName);
        }

        public bool UpdateAgent(Agent agent)
        {
            return this.userRepository.UpdateAgent(agent);
        }

        public bool UpdateUser(User user)
        {
            if (!userValidator.HasValidEmail(user.Email)) return false;

            return this.userRepository.UpdateUser(user);
        }
    }
}

File: src/business/rule/engine/serviceimpl/UserService.java
package business.rule.engine.serviceimpl;

import business.rule.engine.model.Agent;
import business.rule.engine.model.User;
import business.rule.engine.service.IUserRepository;
import business.rule.engine.service.IUserService;

public class UserService implements IUserService
{
    private IUserRepository userRepository;
    private UserValidator userValidator;

    public UserService() {
    	this(new UserRepository());
    }

    public UserService(IUserRepository userRepository)
    {
        this.userRepository = userRepository;
        this.userValidator = new UserValidator();
    }

    public User GetUser(String userName)
    {
        return this.userRepository.GetUser(userName);
    }

    public boolean UpdateAgent(Agent agent)
    {
        return this.userRepository.UpdateAgent(agent);
    }

    public boolean UpdateUser(User user)
    {
        if (!userValidator.HasValidEmail(user.Email)) return false;

        return this.userRepository.UpdateUser(user);
    }   

}
